blogs/views.py

```diff
@@ -0,0 +1,98 @@
+import json
+
+from django.contrib.auth import authenticate, login, logout
+from django.contrib.auth.models import User
+from django.http import JsonResponse
+from django.views.decorators.csrf import ensure_csrf_cookie
+
+from rest_framework import generics
+from rest_framework.views import APIView
+from rest_framework.response import Response
+
+from blogs.models import Blogs, Comment
+from blogs.permissions import IsAuthorOrReadOnly
+from blogs.serializers import BlogsSerializer, CommentSerializer
+
+
+class BlogsList(generics.ListCreateAPIView):
+    """List all blogs, or create a new blog"""
+    queryset = Blogs.objects.all().order_by('-updated_at')
+    serializer_class = BlogsSerializer
+    
+    def get_queryset(self):
+        """For displaying author specific posts if author is specified in url"""
+        return_query = None
+        if self.kwargs:
+            try:
+                author = User.objects.get(username=self.kwargs['author'])
+                return_query = Blogs.objects.filter(author=author).order_by('-updated_at')
+            except:
+                print('Author not found')
+        else:
+            return_query = Blogs.objects.all().order_by('-updated_at')
+        return return_query
+            
+    def perform_create(self, serializer):
+        serializer.save(author=self.request.user)
+
+
+
+
```

Comment:
For quality, try to follow this https://www.python.org/dev/peps/pep-0008/#blank-lines